### PR TITLE
Add gaphical isochrones in navitia-playground

### PR DESCRIPTION
**Pull request by ZoeBrunet**

### CODE REVIEW

File: js/autocomplete.js [hunk 1]
```diff
@@ -48,6 +48,8 @@ var autocomplete = {
             departures: _depArrParams,
             journeys: ['from', 'to', 'datetime', 'datetime_represents', 'traveler_type', 'data_freshness',
             'first_section_mode[]', 'last_section_mode[]'].concat(_paramValueEverywhere).sort(),
+            isochrones: ['from', 'to','datetime', 'traveler_type', 'max_duration','first_section_mode[]',
+            'last_section_mode[]', 'min_duration'].concat(_paramValueEverywhere).sort(),
```
Comment (TeXitoi) [inline js/autocomplete.js]: You'll need to sync with journeys: there is at least `forbidden_uris[]` now. And you should also add `data_freshness`.

File: js/map.js [hunk 2]
```diff
@@ -48,6 +48,11 @@ var map = {
             }
             return flatMap(json.sections, bind);
         },
+        isochrone: function(context, json) {
+            if (! ('geojson' in json)) { return []; }
+              var geoJsonShape = json.geojson;
+              return map._makePolygon(context, 'isochrone', json.geojson, json)
```
Comment (TeXitoi) [inline js/map.js]: bad indent, laking final `;`

File: js/summary.js [hunk 3]
```diff
@@ -105,6 +105,23 @@ summary.make.journey = function(context, json) {
...
 
+summary.make.isochrone = function(context, json) {
+    var res = $('<span>');
+    if ('from' in json) {
+      res.append(sprintf(' from %s, ', htmlEncode(json.from.name)));
+    }
+    if ('to' in json) {
+      res.append(sprintf(' to %s, ', htmlEncode(json.to.name)));
+    }
+    if('min_duration' in json && 'max_duration' in json) {
```
Comment (TeXitoi) [inline js/summary.js]: space between if and (

File: js/summary.js [hunk 4]
```diff
@@ -105,6 +105,23 @@ summary.make.journey = function(context, json) {
...
+        res.append(sprintf('duration: [%s', durationToString(json.min_duration)));
+        res.append(sprintf(', %s]', durationToString(json.max_duration)));
```
Comment (TeXitoi) [inline js/summary.js]: you can do one call to sprintf:
```js
res.append(sprintf('duration: [%s, %s]',
                   durationToString(json.min_duration)
                   durationToString(json.max_duration)));
```

File: js/summary.js [hunk 5]
```diff
@@ -105,6 +105,23 @@ summary.make.journey = function(context, json) {
...
+    } else {
+        res.append('no summary');
```
Comment (TeXitoi) [inline js/summary.js]:
```js
res.text('no summary');
```

File: js/autocomplete.js.orig [hunk 6]
```diff
@@ -0,0 +1,267 @@
+// Copyright (c) 2016 CanalTP
```
Comment (xlqian) [inline js/autocomplete.js.orig]: Ooooops..

File: js/autocomplete.js [hunk 7]
```diff
@@ -125,7 +127,7 @@ var autocomplete = {
             request =  api +  '/coverage/' + cov + '/' + staticType;
         }
         $.ajax({
-            headers: token === undefined ? {} : { Authorization: 'Basic ' + btoa(token) },
+            headers: token ? { Authorization: 'Basic ' + btoa(token) } : {},
```
Comment (xlqian) [inline js/autocomplete.js]: this logic is used in three different places... it's worth being factorized :)

File: js/map.js [hunk 8]
```diff
@@ -58,6 +58,11 @@ var map = {
             }
             return flatMap(json.sections, bind);
         },
+        isochrone: function(context, json) {
+            if (! ('geojson' in json)) { return []; }
+              var geoJsonShape = json.geojson;
```
Comment (xlqian) [inline js/map.js]: unused var

File: js/response.js.orig [hunk 9]
```diff
@@ -0,0 +1,178 @@
+// Copyright (c) 2016 CanalTP
```
Comment (xlqian) [inline js/response.js.orig]: :)

File: js/autocomplete.js (hunk 1 above)
Comment (TeXitoi) [inline js/autocomplete.js]: forbidden_uris[] is in _paramValueEverywhere, but you still need to add data_freshness

File: js/response.js [hunk 10]
```diff
@@ -141,6 +141,11 @@ function Context(data) {
     };
 }
 
+function manage_token (token) {
+  token ? { Authorization: 'Basic ' + btoa(token) } : {}
+  return token
```
Comment (TeXitoi) [inline js/response.js]: just
```js
function manage_token(token) {
    return token ? { Authorization: 'Basic ' + btoa(token) } : {};
}
```

File: js/summary.js [hunk 11]
```diff
@@ -105,6 +105,24 @@ summary.make.journey = function(context, json) {
     return res;
 };
 
+summary.make.isochrone = function(context, json) {
+    var res = $('<span>');
+    if ('from' in json) {
+      res.append(sprintf(' from %s, ', htmlEncode(json.from.name)));
```
Comment (TeXitoi) [inline js/summary.js]: you don't want a space before from and to

File: js/map.js [hunk 12]
```diff
@@ -60,8 +60,8 @@ var map = {
         },
         isochrone: function(context, json) {
             if (! ('geojson' in json)) { return []; }
-              var geoJsonShape = json.geojson;
-              return map._makePolygon(context, 'isochrone', json.geojson, json)
+            var geoJsonShape = json.geojson;
+            return map._makePolygon(context, 'isochrone', geoJsonShape, json);
```
Comment (TeXitoi) [inline js/map.js]: `geoJsonShape` is quite useless, we can give `json.geojson` directly to `_makePolygon`

**Pull request merged**